src/main/java/com/ashkan/moviesapi/services/implementations/MovieServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.ashkan.moviesapi.services.implementations;
+
+import com.ashkan.moviesapi.constants.constants;
+import com.ashkan.moviesapi.repositories.MovieRepository;
+import com.ashkan.moviesapi.repositories.UserRepository;
+import com.ashkan.moviesapi.entities.Movie;
+import com.ashkan.moviesapi.entities.User;
+import com.ashkan.moviesapi.exceptions.NotFound.MovieNotFoundException;
+import com.ashkan.moviesapi.services.interfaces.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+    private MovieRepository movieRepository;
+    private UserRepository userRepository;
+
+    @Autowired
+    public MovieServiceImpl(MovieRepository movieRepository,
+                            UserRepository userRepository) {
+        this.movieRepository = movieRepository;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public List<Movie> findAll() {
```

Comment:
This method is returning a list of Movie Entities, that Entity does not have all the required properties according to the requirements like the **_number of copies_**, maybe you can build a new query to get that information you will need to do a **JOIN** between **Movie** an **Catalog** entities, another option is to build a new query to get the data you need from Catalog Entity and iterate that list to set this value into your result, anyway I think for this requirement you need a totally new object maybe a MovieDTO with the required properties.